File: flask_api/performance_test.py
import time
import requests
import cv2
import numpy as np
from PIL import Image
import io
import os

def test_face_recognition(image_path):
    # Test görüntüsünü yükle
    with open(image_path, 'rb') as f:
        image_data = f.read()
    
    # Yüz tanıma süresini ölç
    start_time = time.time()
    
    # API'ye istek gönder
    response = requests.post(
        'http://localhost:5001/recognize-face',
        files={'image': ('test.jpg', image_data)}
    )
    
    end_time = time.time()
    recognition_time = end_time - start_time
    
    return {
        'status': response.status_code,
        'recognition_time': recognition_time,
        'response': response.json() if response.status_code == 200 else None
    }

def test_face_registration(image_path, user_id, name, surname, angle):
    # Test görüntüsünü yükle
    with open(image_path, 'rb') as f:
        image_data = f.read()
    
    # Yüz kaydetme süresini ölç
    start_time = time.time()
    
    # API'ye istek gönder
    response = requests.post(
        'http://localhost:5001/register-face',
        files={'image': ('test.jpg', image_data)},
        data={
            'userId': user_id,
            'name': name,
            'surname': surname,
            'angle': angle
        }
    )
    
    end_time = time.time()
    registration_time = end_time - start_time
    
    return {
        'status': response.status_code,
        'registration_time': registration_time,
        'response': response.json() if response.status_code == 200 else None
    }

def run_performance_test():
    print("Performans Testi Başlıyor...")
    print("-" * 50)
    
    # Test görüntüsü yolu
    test_image = "test_face.jpg"  # Test için bir yüz görüntüsü kullanın
    
    if not os.path.exists(test_image):
        print(f"Hata: {test_image} bulunamadı!")
        return
    
    # Yüz tanıma testi
    print("\nYüz Tanıma Testi:")
    recognition_result = test_face_recognition(test_image)
    print(f"Tanıma Süresi: {recognition_result['recognition_time']:.2f} saniye")
    print(f"Durum Kodu: {recognition_result['status']}")
    if recognition_result['response']:
        print(f"Yanıt: {recognition_result['response']}")
    
    # Yüz kaydetme testi
    print("\nYüz Kaydetme Testi:")
    registration_result = test_face_registration(
        test_image,
        "test_user",
        "Test",
        "User",
        "front"
    )
    print(f"Kaydetme Süresi: {registration_result['registration_time']:.2f} saniye")
    print(f"Durum Kodu: {registration_result['status']}")
    if registration_result['response']:
        print(f"Yanıt: {registration_result['response']}")

if __name__ == "__main__":
    run_performance_test() 

File: face-gateway/src/performance-test.ts
import { Test } from '@nestjs/testing';
import { INestApplication } from '@nestjs/common';
import supertest from 'supertest';
import { AppModule } from './app.module';
import * as fs from 'fs';
import * as path from 'path';

describe('Face Recognition Performance Test', () => {
  let app: INestApplication;
  const testImagePath = path.join(__dirname, 'test_face.jpg');

  beforeAll(async () => {
    const moduleRef = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = moduleRef.createNestApplication();
    await app.init();

    // Test görüntüsünün varlığını kontrol et
    if (!fs.existsSync(testImagePath)) {
      console.error(`Test görüntüsü bulunamadı: ${testImagePath}`);
      process.exit(1);
    }
  });

  afterAll(async () => {
    await app.close();
  });

  it('should measure face registration performance', async () => {
    console.log('\nYüz Kaydetme Performans Testi:');
    console.log('-'.repeat(50));

    const startTime = Date.now();

    const response = await supertest(app.getHttpServer())
      .post('/face/register')
      .attach('image', testImagePath)
      .field('userId', 'test_user')
      .field('name', 'Test')
      .field('surname', 'User')
      .field('angle', 'front');

    const endTime = Date.now();
    const duration = (endTime - startTime) / 1000; // saniye cinsinden

    console.log(`Kaydetme Süresi: ${duration.toFixed(2)} saniye`);
    console.log(`Durum Kodu: ${response.status}`);
    console.log('Yanıt:', response.body);
  });

  it('should measure face recognition performance', async () => {
    console.log('\nYüz Tanıma Performans Testi:');
    console.log('-'.repeat(50));

    const startTime = Date.now();

    const response = await supertest(app.getHttpServer())
      .post('/face/recognize')
      .attach('image', testImagePath);

    const endTime = Date.now();
    const duration = (endTime - startTime) / 1000; // saniye cinsinden

    console.log(`Tanıma Süresi: ${duration.toFixed(2)} saniye`);
    console.log(`Durum Kodu: ${response.status}`);
    console.log('Yanıt:', response.body);
  });
}); 